core/math/src/main/java/org/arakhne/afc/math/geometry/d2/ai/Rectangle2ai.java

```diff
@@ -375,7 +375,7 @@ default double getDistanceSquared(Point2D<?, ?> pt) {
 		} else {
 			dy = 0;
 		}
-		return dx * dx + dy * dy;
+		return dx * dx + (double) dy * dy;
```

Comment:
Not good because you force the usage of the double addition instead using the addition between integer before casting the resulting integer into a double.